src/libpsl-native/src/getcurrentthreadid.cpp

```diff
@@ -3,14 +3,21 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/syscall.h>
+#include <pthread.h>
 
 pid_t GetCurrentThreadId()
 {
     pid_t tid = 0;
 #if defined(__linux__)
     tid = syscall(SYS_gettid);
 #elif defined(__APPLE__) && defined(__MACH__)
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12
+    uint64_t tid64;
+    pthread_threadid_np(NULL, &tid64);
```

Comment:
I'm fine with this change (and indeed it will be necessary when .NET Core moves to 10.12), but we don't need to conditionalize it. Please make it the only code on macOS.